sormas-backend/src/test/java/de/symeda/sormas/backend/person/PersonFacadeEjbTest.java

```diff
@@ -1778,4 +1778,128 @@ public void testIsEditAllowedLinkedToNotOwnedCaseAndDeletedAssociatedObject() {
 		getImmunizationFacade().delete(immunization.getUuid(), new DeletionDetails());
 		assertFalse(getPersonFacade().isEditAllowed(person.getUuid()));
 	}
+
+	@Test
+	public void testIsSharedOrReceived() {
+		RDCF rdcf = creator.createRDCF();
+		UserDto userDto = creator.createUser(rdcf, creator.getUserRoleReference(DefaultUserRole.NATIONAL_USER));
+		User user = getUserService().getByUuid(userDto.getUuid());
+
+		PersonDto person = creator.createPerson();
+		CaseDataDto sharedCase = creator.createCase(userDto.toReference(), person.toReference(), rdcf, null);
+
+		assertFalse(getPersonFacade().isSharedOrReceived(person.getUuid()));
+
+		// shared case
+		creator.createShareRequestInfo(
+			ShareRequestDataType.CASE,
+			user,
+			"target_id",
+			false,
+			ShareRequestStatus.ACCEPTED,
+			(s) -> s.setCaze(getCaseService().getByReferenceDto(sharedCase.toReference())));
+
+		assertTrue(getPersonFacade().isSharedOrReceived(person.getUuid()));
+
+		// received case
+		person = creator.createPerson();
```

Comment:
general remark: I think it would be beneficial to crate a `creator.createReceivedCase` etc method by now? It is frequently encountered.